### PR TITLE
Allow to create graph space without options

### LINKED ISSUE #350: CREATE SPACE should have default options
e.g. `CREATE SPACE myspace` shall succeed.

### PR DESCRIPTION
**Pull request by dutor**
As title.

This PR will close #350

### CODE REVIEW

File: src/parser/AdminSentences.h [hunk 1]
```diff
@@ -223,6 +223,9 @@ class CreateSpaceSentence final : public Sentence {
     }
 
     std::vector<SpaceOptItem*> getOpts() {
+        if (spaceOpts_ == nullptr) {
+            return {};
+        }
```
Comment (steppenwolfyuetong) [inline src/parser/AdminSentences.h]: when create graph space without options, CreateSpaceSentence::toString function need to  deal with

File: src/graph/test/SchemaTest.cpp [hunk 2]
```diff
@@ -319,6 +319,20 @@ TEST_F(SchemaTest, metaCommunication) {
...
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    sleep(FLAGS_load_data_interval_secs + 1);
+    {
+        // TODO To verify the space options are applied correctly.
+        cpp2::ExecutionResponse resp;
+        std::string query = "USE space_with_default_options";
+        auto code = client->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
```
Comment (steppenwolfyuetong) [inline src/graph/test/SchemaTest.cpp]: Here use `describe space_with_default_options` to handle more obvious

File: src/parser/AdminSentences.h (hunk 1 above)
Comment (dutor) [reply src/parser/AdminSentences.h]: roger that!

File: src/graph/test/SchemaTest.cpp (hunk 2 above)
Comment (dutor) [reply src/graph/test/SchemaTest.cpp]: Yes, we will do, once #388 is merged.

File: src/interface/meta.thrift [hunk 3]
```diff
@@ -88,8 +88,8 @@ struct ExecResp {
 // Graph space related operations.
 struct CreateSpaceReq {
     1: string space_name,
-    2: i32 parts_num,
-    3: i32 replica_factor,
+    2: optional i32 parts_num,
```
Comment (dangleptr) [inline src/interface/meta.thrift]: Default field is enough in this case.

Comment (dutor) [reply src/interface/meta.thrift]: I'm not sure whether it's OK if a non-optional field is not set, is that allowed?

Anyway, the present implementation of meta client set this field blindly. There is no such thing _default_ or _optional_.

Maybe this is a special case where there are values could be used as a default to represent NULL or not set. But we must be careful about cases there aren't such values.

The thrift proto on this struct in upstream has been modified, so this change should be discarded.

**Comment by dutor** [comment]
Jenkins, go!

**Comment by dutor** [comment]
Jenkins go

**Comment by dutor** [comment]
Jenkins go

**Comment by dutor** [comment]
Jenkins go

**Pull request merged**